icu4c/source/common/locid.cpp

```diff
@@ -1357,7 +1361,9 @@ Locale::setKeywordValue(const char* keywordName, const char* keywordValue, UErro
         }
         fullName = newFullName;
         status = U_ZERO_ERROR;
-        newLength = uloc_setKeywordValue(keywordName, keywordValue, fullName, newLength, &status) + 1;
+        uloc_setKeywordValue(keywordName, keywordValue, fullName, newLength, &status) + 1;
+    } else {
+        U_ASSERT(status != U_BUFFER_OVERFLOW_ERROR);
```

Comment:
This lines makes no sense, after `if (status == U_BUFFER_OVERFLOW_ERROR)` it couldn't ever happen that this assertion isn't true, unless you have a compiler so broken that it generates bad code for if-statements.